.github/workflows/auto-set-base.yml

```diff
@@ -0,0 +1,23 @@
+name: Auto-set Base Branch to Develop
+
+on:
+  pull_request:
+    types: [opened]
+
+jobs:
+  set-base-branch:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Check and Update Base Branch
+        run: |
+          if [ "${{ github.event.pull_request.base.ref }}" == "main" ]; then
```

Comment:
We should have the logic be 
if TARGET_BRANCH == main && BASE_BRANCH != develop
    switch to develop